### PR TITLE
netvfy provisioning bug fix

### PR DESCRIPTION
**Pull request by si74**

### CODE REVIEW

File: src/ndb.c [hunk 1]
```diff
@@ -367,6 +367,16 @@ ndb_provisioning(const char *provlink, const char *network_name)
 	const char			*cacert;
 	const char			*cert;
 
+	if (strcmp(provlink, "") == 0) {
+		fprintf(stderr, "%s: provisioning key must be defined \n", __func__);
```
Comment (nicboul) [inline src/ndb.c]: nit: the \n is usually next to the last word. e.g: [...]defined\n"

File: src/ndb.c [hunk 2]
```diff
@@ -396,17 +406,22 @@ ndb_provisioning(const char *provlink, const char *network_name)
...
 		return (-1);
+	}
 
 	if (((version = evhttp_find_header(&headers, "v")) == NULL) ||
 	    ((provsrv_addr = evhttp_find_header(&headers, "a")) == NULL) ) {
+		fprintf(stderr, "%s: invalid provisioning key \n", __func__); 
 		return (-1);
 	}
-
+
```
Comment (nicboul) [inline src/ndb.c]: seems a tab have been involuntarily added here

File: src/ndb.c [hunk 3]
```diff
@@ -422,6 +437,9 @@ ndb_provisioning(const char *provlink, const char *network_name)
 	curl_global_init(CURL_GLOBAL_ALL);
 
 	/* get a curl handle */
+
+	// TODO(sneha): have a more reasonable timeout. Occasionally this command will hang a very long time
+	// with an incorrect provisioning key.
```
Comment (nicboul) [inline src/ndb.c]: Also I wonder if the controller return promptly if a provisioning key is invalid. The timeout should be 5seconds (curl_easy_setopt(curl, CURLOPT_TIMEOUT, 5L)). I'm curious why it doesn't behave properly.

File: src/ndb.c [hunk 4]
```diff
@@ -477,4 +495,5 @@ ndb_provisioning(const char *provlink, const char *network_name)
 	return (0);
 }
 
+
```
Comment (nicboul) [inline src/ndb.c]: seems a new line have been involuntarily been added here

Comment (si74) [reply src/ndb.c]: fixed

File: src/ndb.c (hunk 3 above)
Comment (si74) [reply src/ndb.c]: Yes. So I tested further and most invalid provisioning keys are returning promptly...However, I am able to connect with incorrect keys (those with several characters deleted off the end but the URI for the most part having the expected keys). See notes from here: https://docs.google.com/document/d/1rf4jOnzSQF8EE2JkCQsVMrBcashqNmsuhx06tGDLfwM/edit

File: src/ndb.c (hunk 2 above)
Comment (si74) [reply src/ndb.c]: fixed

**Review by nicboul — APPROVED** [review]
LGTM !

**Pull request merged**